common/protocol/jest.config.ts

```diff
@@ -1,16 +1,25 @@
-import type { JestConfigWithTsJest } from "ts-jest";
+import type { JestConfigWithTsJest } from 'ts-jest';
 
 const config: JestConfigWithTsJest = {
-  preset: "ts-jest/presets/default-esm",
-  testEnvironment: "node",
-  extensionsToTreatAsEsm: [".ts"],
-  transform: {
-    "^.+\\.ts$": ["ts-jest", { useESM: true }],
-  },
-  moduleNameMapper: {
-    "^(\\.{1,2}/.*)\\.js$": "$1",
-  },
-  testPathIgnorePatterns: ["/dist/"],
+    preset: 'ts-jest/presets/default-esm',
+    testEnvironment: 'node',
+    extensionsToTreatAsEsm: ['.ts'],
+    transform: {
+        '^.+\\.ts$': ['ts-jest', { useESM: true }],
+        '^.+\\.js$': ['ts-jest', { useESM: true }],
+    },
+    transformIgnorePatterns: [
+        'node_modules/(?!(@filoz/(synapse-sdk|synapse-core)|@web3-storage|multiformats|sync-multihash-sha2|iso-web|delay|p-retry|is-network-error|dnum))',
```

Comment:
could be extracted to helper function with array of package names you want to ensure are transformed